### PR TITLE
cleanup dead installer code

### PR DESCRIPTION
**Pull request by deads2k**
cleans up some installer code that was dead and reorders a little of the flow to reduce complexity.

@kubernetes/sig-api-machinery-misc

### CODE REVIEW

**Comment by sttts** [comment]
There is also the unused `storage` map in GenericApiServer as @frobware noticed. If you are at it, can you remove that as well?

**Comment by deads2k** [comment]
> There is also the unused storage map in GenericApiServer as @frobware noticed. If you are at it, can you remove that as well?

done

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -85,8 +85,10 @@ var toDiscoveryKubeVerb = map[string]string{
 }
 
 // Install handlers for API resources.
-func (a *APIInstaller) Install(ws *restful.WebService) (apiResources []metav1.APIResource, errors []error) {
-	errors = make([]error, 0)
+func (a *APIInstaller) Install() ([]metav1.APIResource, *restful.WebService, []error) {
+	var apiResources []metav1.APIResource
+	ws := a.newWebService()
+	errors := make([]error, 0)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: nit: var errors []error

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 2]
```diff
@@ -112,11 +114,11 @@ func (a *APIInstaller) Install(ws *restful.WebService) (apiResources []metav1.AP
 			apiResources = append(apiResources, *apiResource)
 		}
 	}
-	return apiResources, errors
+	return apiResources, ws, errors
 }
 
 // NewWebService creates a new restful webservice with the api installer's prefix and version.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: newWebService

**Comment by sttts** [comment]
nits. Lgtm otherwise.

**Pull request merged**